Source/main.cpp

```diff
@@ -5,9 +5,17 @@ int width = 1200;
 int height = 720;
 
 int main(int argc, char* argv[]) {
-	Engine engine(width, height, DEBUG_DRAWTIME);
+	Engine engine(width, height);
 
-	Mesh mesh(100, 40, 50);
+	TTF_Font* mono = TTF_OpenFont("./Assets/FreeMono.ttf", 15);
+
+	UIText text("Test Text");
+
+	text.setRenderer(engine.getRenderer());
```

Comment:
For ease of use, and to make `Engine`'s implementation a little more opaque, I'd move this step into `addUIObject` and allow `UIText` (or other `UIObject`s) to be created, configured, and added freely without needing to explicitly acquire the renderer.

```
void Engine::addUIObject(UIObject* uiObject) {
  uiObject.setRenderer(renderer);

  ui->addObject(object);
}
```

Reply:
Fixed!